Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2881 -- Remove redundant list on subject homepage

Resolves [EREGCSC-2881](https://jiraent.cms.gov/browse/EREGCSC-2881)

**Description**

The Subjects list in the center of the Subject page, known as the "Table of Contents", is no longer needed: it duplicates the functionality of the Subjects Selector list that now exists in the left sidebar (see #1487).  

This PR will remove the Table of Contents and replace it with copy text about how to use the Subjects page, which will be referred to as the "Subject Landing page". 

**This pull request changes:**

- Renames `SubjectTOC.vue` to `SubjectLanding.vue`
   - Removes Subject list
   - changes `h1` heading to read "Find Policy Documents"
   - adds copy text (in the form of a `ul`) and link to `/about/`
   - Behavior about when Subject Landing page is shown is the same as before
- Adds ability for user to reset all selected filters on Subjects page by clicking "Research a Subject" header link
   - Adds custom event `link-clicked` to Header Links that is emitted on click with current page name as payload
   - Adds event listener (via `v-on` directive's `@` shorthand) in `Subjects.vue` that listens for this emitted event
   - On "Research a Subject" click, all query params are removed and Subject Landing page is shown
- Removes bold font weight from left sidebar Subject subtitle on focus/hover
- Tidies up focus and focus-visible outlines in Subject Selector and in a few buttons around the app

**Steps to manually verify this change:**

1. Green check marks
2. Visit `/subjects/` on the [experimental deployment](https://g6a6u083x0.execute-api.us-east-1.amazonaws.com/dev1494/subjects/)
3. Ensure that you see the Subject Landing page in the center of the page and not a duplicated list of subjects
4. Add and remove subjects and other filters to ensure page works as expected
5. When you have several filters selected, click "Research a Subject" at the top right of the header to reset all filters and once again show the Subject Landing page
6. When hovering the cursor over the list of subjects in the left sidebar, no subject subtitle should have a bold style applied
